### PR TITLE
Limit and stop orders

### LINKED ISSUE #1245: Implement limit orders for cross margin accounts

### LINKED ISSUE #1394: Calculate minimum margin in deposit modal
When using the deposit margin modal, currently it checks that the minimum value of deposit is 50 sUSD. Input values below 50 sUSD currently disable the button, ignoring already deposited margin. 

Users might already have margin deposited in a market, and would only want to top up to reach 50$. Currently users are forced to deposit minimum 50$ by default. 

This should be changed to set the button to only disable if the deposit + current margin is <$50. 

### Deliverables

- [ ] Update deposit modal for Futures markets to disable the button only if `(deposit + current margin) < 50$`

### Rewards
100 $OP

### PR DESCRIPTION
**Pull request by avclarke**
## Description
Adds the ability to submit stop and limit orders on cross margin futures positions

## Related issue
closes #1245
closes #1394

## Tasks

- [x] Submit limit order
- [x] Submit stop order
- [x] Update fees to show keeper deposit amount
- [x] Update trade preview to account for dynamic order price
- [x] Update trade confirmation to show order type and price 
- [x] Add validation on order price input
- [x] Display keeper deposit and withdraw
- [x] Update order size input to toggle between asset types
- [x] Fee rejection options
- [x] Include orders fees in fee info
- [x] Cancel limit / stop orders

### CODE REVIEW

File: sections/futures/TradeCrossMargin/CrossMarginInfoBox.tsx [hunk 1]
```diff
@@ -164,12 +178,27 @@ function MarginInfoBox({ editingLeverage }: Props) {
...
 						),
 					},
+					'Keeper ETH Deposit':
+						orderType === 'limit' || orderType === 'stop'
+							? {
+									value: formatCurrency('ETH', keeperEthBal) + 'Ξ',
+									valueNode: (
+										<>
+											{keeperEthBal.gt(0) && (
+												<ActionButton hideBorder onClick={() => setOpenModal('keeper-deposit')}>
```
Comment (Tburm) [inline sections/futures/TradeCrossMargin/CrossMarginInfoBox.tsx:188]: Should we have a similar "deposit" arrow? As I user I could do the following:
* Open first limit order, which deposits Keeper ETH
* Withdraw my ETH
* Now my orders won't execute, and I can't deposit ETH without a new order

File: sections/futures/TradeCrossMargin/CrossMarginInfoBox.tsx [hunk 2]
```diff
@@ -26,23 +30,28 @@ import {
...
 
 import EditLeverageModal from './EditLeverageModal';
+import WithdrawKeeperDepositModal from './WithdrawKeeperDepositModal';
 
 type Props = {
 	editingLeverage?: boolean;
 };
 
 function MarginInfoBox({ editingLeverage }: Props) {
+	const { keeperEthBal } = useCrossMarginKeeperDeposit();
```
Comment (Tburm) [inline sections/futures/TradeCrossMargin/CrossMarginInfoBox.tsx:40]: Doesn't look like this value is refetched when things change. We should fetch it if a user submits an order which deposits ETH, or if they withdraw their ETH.

**Review by Tburm — COMMENTED** [review]
Tested all major functions submitting limit and stop orders. Left comments on things we should fix in this PR, and I'll submit some other feedback separately to follow up.

It works really well, great work!

File: sections/futures/Trade/TradeConfirmationModal.tsx [hunk 3]
```diff
@@ -39,13 +46,25 @@ export default function TradeConfirmationModal({
...
+	const positionSide = useMemo(() => {
+		if (potentialTradeDetails?.size.eq(zeroBN)) {
+			return position?.position?.side === PositionSide.LONG
+				? PositionSide.SHORT
+				: PositionSide.LONG;
+		}
+		return potentialTradeDetails?.size.gte(zeroBN) ? PositionSide.LONG : PositionSide.SHORT;
+	}, [potentialTradeDetails, position?.position?.side]);
 
 	const positionDetails = useMemo(() => {
```
Comment (Tburm) [inline sections/futures/Trade/TradeConfirmationModal.tsx:62]: The `protocolFee` value does not include the stop/limit order fee. The modal displays protocol fee which only includes the trading fee and cross margin fee.

File: sections/futures/Trade/TradeConfirmationModal.tsx [hunk 4]
```diff
@@ -39,13 +46,25 @@ export default function TradeConfirmationModal({
...
 		return potentialTradeDetails
 			? {
 					...potentialTradeDetails,
 					size: potentialTradeDetails.size.abs(),
```
[the remark this answers is not in the document]
Comment (avclarke) [reply sections/futures/Trade/TradeConfirmationModal.tsx:66]: Good call, will update to use sizeDelta

File: sections/futures/TradeCrossMargin/CrossMarginInfoBox.tsx (hunk 1 above)
Comment (avclarke) [reply sections/futures/TradeCrossMargin/CrossMarginInfoBox.tsx:187]: I've built the UI here but commented out for now until we have a deposit function on the smart contract.

**Review by platschi — APPROVED** [review]

**Review by Tburm — APPROVED** [review]

**Pull request merged**